core/shared/src/main/scala/cats/parse/LocationMap.scala

```diff
@@ -91,3 +93,13 @@ class LocationMap(val input: String) {
 object LocationMap {
   def apply(str: String): LocationMap = new LocationMap(str)
 }
+
+case class Caret(offset: Int, row: Int, col: Int)
+object Caret {
+  implicit val eqCatsCaret: Eq[Caret] = Eq.fromUniversalEquals
```

Comment:
what about making an `Order` instead and order by the `offset`? We might want to sort by carets.